specification/dsdl/serializable_types.tex

```diff
@@ -281,51 +313,282 @@ \subsubsection{Dependencies}
 
+To avoid ambiguity, a data type that is not a tagged union is referred to as a \emph{structure}.
+
+\subsubsection{Alignment and cumulative bit length set}\label{sec:dsdl_composite_alignment_cumulative_bls}
+
+The alignment of composite types is one byte (8 bits)\footnote{%
+    Regardless of the content.
+    It follows that empty composites can be inserted arbitrarily to force byte alignment
+    of the next field(s) at runtime.
+}.
+
+Per the definitions given in \ref{sec:dsdl_serializable_alignment_padding},
+a serialized representation of a composite type is padded to 8 bits by inserting padding bits
+after its last element until the resulting length is a multiple of 8 bits.
+
+Given a set of field attributes, their \emph{cumulative bit length set} is computed
+by evaluating every permutation of their respective bit length sets plus the required padding.
+\begin{itemize}
+    \item For tagged unions, this amounts to the union of the bit length sets of each field
+    plus the bit length set of the implicit union tag prefix.
+    \item Otherwise, the cumulative bit length set is the Cartesian product of the bit length sets of each field
+    plus the required inter-field padding.
+\end{itemize}
+Related specifics are given in section \ref{sec:dsdl_data_serialization} on data serialization.
+
+\subsubsection{Extent and finality}\label{sec:dsdl_composite_extent_and_finality}
+
+As detailed in section \ref{sec:dsdl_versioning},
+data types may evolve over time to accommodate new design requirements, new features, to rectify issues, etc.
+In order to allow gradual migration of deployed systems to revised data types,
+it is desirable to ensure that they can be modified in a way that does not render new definitions
+incompatible with their earlier versions.
+In this context there are two related concepts:
+
+\begin{description}
+    \item[Extent] --- the minimum amount of memory, in bits, that shall be allocated to store the serialized
+    representation of the type.
+    The extent of any type is always greater than or equal the maximal value of its bit length set.
+    It is always a multiple of 8.
+
+    \item[Finality] --- denotes whether a type is \emph{final}.
+    A type that is final is non-evolvable and its extent equals the maximal value of its bit length set\footnote{%
+        I.e., the smallest possible extent.
+    }.
+    A type that is not final is also referred to as \emph{delimited}.
+\end{description}
+
+The extent is the growth limit for the maximal bit length of the type as it evolves.
+The extent should be at least as large as the longest serialized representation of any compatible version of the type,
+which ensures that an agent leveraging a particular version can correctly process any other compatible
+version due to the avoidance of unintended truncation of serialized representations.
+
+Serialized representations of evolvable definitions carry additional metadata which introduces a certain overhead.
```

Comment:
But only if they are nested, right? There isn't additional meta-data for top-level types.